### ISSUE TITLE
IDE0037 is case insensitive for VB

### ISSUE THREAD

**Issue by doesicool** [dotnet/roslyn#84278]
The code style rule IDE0037: 'Use inferred member names' doesn't correctly work for VB (but works for C#). It erroneously reports expressions with different casing. Take a look at code samples below:
### VB
```vb
Public NotInheritable Class TestClass
	Public Shared Sub Test1()
		Dim Bar As Integer = 1

		' Correct: warning IDE0037: Member name can be simplified
		Dim foo As New With {.Bar = Bar}
	End Sub

	Public Shared Sub Test2()
		Dim bar As Integer = 2

		' False positive warning IDE0037: Member name can be simplified
		Dim foo As New With {.Bar = bar}
	End Sub

	Public Shared Sub Test3()
		Dim bAr As Integer = 3

		' False positive warning IDE0037: Member name can be simplified
		Dim foo As New With {.Bar = bAr}
	End Sub
End Class
```
### C#
```cs
public sealed class TestClass
{
	public static void Test1()
	{
		int Bar = 1;

		// Correct: warning IDE0037: Member name can be simplified
		var foo = new { Bar = Bar };
	}

	public static void Test2()
	{
		int bar = 2;

		// Correct: No warning IDE0037: Member name can be simplified
		var foo = new { Bar = bar };
	}

	public static void Test3()
	{
		int bAr = 3;

		// Correct: No warning IDE0037: Member name can be simplified
		var foo = new { Bar = bAr };
	}
}
```
There is an old related issue [#23659](https://github.com/dotnet/roslyn/issues/23659#issue-280345535) with an [comment](https://github.com/dotnet/roslyn/issues/23659#issuecomment-351009292) which also adresses this problem.

(Additional Information: Tested with Visual Studio 2026 Version 18.7.2 in .NET 8 class libraries with &lt;AnalysisLevel&gt; = preview-all.)

**Comment by CyrusNajmabadi** [comment]
I'm not understanding the problem here, with VB being case insensitive, this change seems reasonable to offer.  Can you clarify what the concern is here?

**Comment by doesicool** [comment]
As in the referenced [comment](https://github.com/dotnet/roslyn/issues/23659#issuecomment-351009292) mentioned different casing lead to different anonymous objects. E.g. if you serialize them you get different results. So firing IDE0037 is wrong.

**Comment by CyrusNajmabadi** [comment]
> E.g. if you serialize them you get different results. So firing IDE0037 is wrong.

Serialization it out of bounds of what the language and tooling like this care about.  Virtually all our features can break once things like reflection get involved.  If you actually need explicit contract with external systems that are outside the lang, then you should disable this analyzer.

**Comment by doesicool** [comment]
Serialization was just an example where this can lead to unexpected behavior. I know that VB generally is case insensitive but i wasn't aware of that even anonymous types are created in a case insensitive manner by the VB compiler. Consider the following example:
```vb
Public Sub Main()
	Dim anonymousObj2_bAr = New With {.bAr = Nothing}
	Dim anonymousObj1_Bar = New With {.Bar = Nothing}

	Dim t1_Bar = anonymousObj1_Bar.GetType()
	Dim t2_bAr = anonymousObj2_bAr.GetType()
	Dim areTypesEqual = t1_Bar = t2_bAr ' = True

	Dim propertyName = t1_Bar.GetProperties.Single.Name
	' = "Bar" if object 1 is declared first
	' = "bAr" if object 2 is declared first
End Sub
```
For both anonymous objects the compiler creates the same anonymous type. However the name of the created property depends on the order of the declaration. There are two references in the documentation ([1](https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/objects-and-classes/anonymous-types#:~:text=Name%20comparisons%20are%20not%20case%2Dsensitive), [2](https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/objects-and-classes/anonymous-type-definition#:~:text=Name%20comparisons%20are%20not%20case%2Dsensitive)) that state _„Name comparisons are not case-sensitive.“_ regarding equality but not about how properties are generally named. I also was just confused about the fact that Visual Studio uses some magic and shows different casing in intellisense even though the underlying anonymous type is the same.

So i understand now why IDE0037 is case insensitive in VB and why it gets fired in all three cases in the initial post. Hence it's not an issue and i'm closing it out.
